### ISSUE TITLE
Android compilation issues

### ISSUE THREAD

**Issue by eduardodoria** [erincatto/box2d#782]
I had to modify some parts to build for Android.

In file `allocate.c` the function `aligned_alloc` is only supported starting from Android API level 28, so I'm using `posix_memalign` instead:

```
#if defined(B2_PLATFORM_WINDOWS)
    void* ptr = _aligned_malloc( size32, B2_ALIGNMENT );
#elif defined(B2_PLATFORM_ANDROID)
    void* ptr = NULL;
    if (posix_memalign(&ptr, B2_ALIGNMENT, size32) != 0) {
        ptr = NULL;  // Allocation failed
    }
#else
    void* ptr = aligned_alloc( B2_ALIGNMENT, size32 );
#endif
```

In file `core.h` I modified this part to compile for x86 and arm (maybe need to rename `B2_CPU_X64` macro):

```
// Define CPU
#if defined( __x86_64__ ) || defined( _M_X64 )
#define B2_CPU_X64
#elif defined( __i386__ ) || defined( _M_IX86 )
#define B2_CPU_X64
#elif defined( __aarch64__ ) || defined( _M_ARM64 )
#define B2_CPU_ARM
#elif defined( __arm__ ) || defined( _M_ARM )
#define B2_CPU_ARM
#elif defined( __EMSCRIPTEN__ )
#define B2_CPU_WASM
#else
#error Unsupported CPU
#endif
```

At same `core.h` file I needed to add `B2_BREAKPOINT` macro to Android:

```
#if defined( B2_COMPILER_MSVC )
#define B2_BREAKPOINT __debugbreak()
[...]
#elif defined( B2_PLATFORM_ANDROID )
#define B2_BREAKPOINT __builtin_trap()
[...]
#else
#error Unknown platform
#endif
```

**Comment by erincatto** [comment]
Thanks for the write up.

Shouldn't the application exit if it is out of memory?

Isn't the break point a property of the compiler?

**Comment by erincatto** [comment]
Fixed in #783
